benchmarks-readme.md

```diff
@@ -1,29 +1,43 @@
+Hostnames are configured via the `scripts/benchmarks/servers.txt` file.
+The format is mostly self-explanatory; there are spots to configure the client node, the three timelock cluster nodes, and three cassandra nodes.
+Finally, there is a variable for the keystore password; see below for details.
 
-## Host Setup
-The scripts require your user to have access to the `/opt/palantir/timelock` directory, and to be able to view running processes. This applies to both the benchmark cluster and client hosts.
+## Cassandra Setup & SSL
+Your cassandra cluster should be configured to require client ssl.
+The user should provide a keystore and a truststore to be used with cassandra and store them in the `scripts/benchmarks` folder.
+The keystore password should be substituted in the `servers.txt` file mentioned above.
+The suggested way to configure this is to use cassandra's keystore and the truststore, or use the certs from another service that was already configured.
 
-## Cassandra Setup
-Your cassandra cluster should be configured to NOT require client ssl. This is set in `cassandra.yaml` via the `client_encryption_options.enabled` property (should be set to `false`). Alternatively you can enable ssl, but you'll need to modify the atlas configuration in `timelock-server-benchmark-client/var/conf/template/benchmark-server.yml` to use ssl for cassandra, and create a truststore.
+## Host Setup
+The scripts require your user to have access to the `/tmp/timelock-benchmark` directory, and to be able to view running processes.
+This applies to both the benchmark cluster and client hosts.
 
 ## Running Benchmarks
 To run the benchmarks:
-1. Launch four server boxes (3 for the timelock cluster + 1 for the benchmark client), and configure them per the "host setup" section above
-2. Launch a cassandra cluster, and configure it per the "cassandra setup" section above
-3. Add the hostnames to `scripts/benchmarks/servers.txt`
-4. Run `./push-benchmark-client` and `./push-benchmark-cluster` to deploy the client and timelock cluster
-5. Use `BenchmarksRunner.java` to execute benchmarks
+1. Launch a cassandra cluster, and configure it per the "cassandra setup" section above
+2. Launch four server boxes (3 for the timelock cluster + 1 for the benchmark client), and configure them per the "host setup" section above
+3. Add the hostnames and keystore password to `scripts/benchmarks/servers.txt`
+4. Run `./push-benchmark-cluster` and `./push-benchmark-client` to deploy the client and timelock cluster
+5. Use `BenchmarksRunner.java` or curl the client to execute benchmarks.
+
+For example, one can execute the following on the machine that is running the client:
+`curl -kX GET "http://localhost:9425/perf/contended-write-txn?numClients=1&numRequestsPerClient=1"`
```

Comment:
nit: We really don't need the options, as `-k` isn't needed for HTTP, and `-XGET` is the default behaviour.